dali/operators/reader/fits_reader_op.cc

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2020-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+
+#include "dali/core/backend_tags.h"
+#include "dali/core/static_switch.h"
+#include "dali/kernels/slice/slice_cpu.h"
+#include "dali/kernels/slice/slice_flip_normalize_permute_pad_cpu.h"
+#include "dali/kernels/transpose/transpose.h"
+#include "dali/operators/reader/fits_reader_op.h"
+#include "dali/pipeline/data/backend.h"
+
+namespace dali {
+namespace detail {
+inline int FitsReaderOutputFn(const OpSpec &spec) {
+  // there is default value provided, so no checking if arg exists is needed
+  return spec.GetRepeatedArgument<int>("hdu_indices").size();
+}
+
+}  // namespace detail
+
+DALI_REGISTER_OPERATOR(experimental__readers__Fits, FitsReaderCPU, CPU);
+
+DALI_SCHEMA(experimental__readers__Fits)
+    .DocStr(R"(Reads Fits image HDUs from a directory.
+
+This operator can be used in the following modes:
+
+1. Read all files from a directory indicated by ``file_root`` that match given ``file_filter``.
+2. Read file names from a text file indicated in ``file_list`` argument.
+3. Read files listed in ``files`` argument.
+)")
+    .NumInput(0)
+    .OutputFn(detail::FitsReaderOutputFn)
+    .AddOptionalArg<string>("file_root",
+                            R"(Path to a directory that contains the data files.
+
+If not using ``file_list`` or ``files``, this directory is traversed to discover the files.
+``file_root`` is required in this mode of operation.)",
+                            nullptr)
+    .AddOptionalArg(
+        "file_filter",
+        R"(If a value is specified, the string is interpreted as glob string to filter the
+list of files in the sub-directories of the ``file_root``.
+
+This argument is ignored when file paths are taken from ``file_list`` or ``files``.)",
+        "*.fits")
+    .AddOptionalArg<string>("file_list",
+                            R"(Path to a text file that contains filenames (one per line)
+where the filenames are relative to the location of that file or to ``file_root``, if specified.
```

Comment:
nitpick. I'd split that sentence in two for clarity.

```suggestion
                            R"(Path to a text file that contains filenames (one per line).
The filenames are relative to the location of that file or to ``file_root``, if specified.
```

Reply:
Ok, all suggestions regarding spec were applied.